src/main/java/de/tum/in/www1/artemis/web/rest/StudentQuestionResource.java

```diff
@@ -67,24 +67,24 @@ public StudentQuestionResource(StudentQuestionRepository studentQuestionReposito
      *
      * @param courseId course the question belongs to
      * @param studentQuestion the studentQuestion to create
-     * @return the ResponseEntity with status 201 (Created) and with body the new studentQuestion, or with status 400 (Bad Request) if the studentQuestion has already an ID
+     * @return the ResponseEntity with status 201 (Created) and with body the new studentQuestion, or with status 400 (Bad Request) if the studentQuestion
+     * already
+     * has an ID or the courseId in the body doesnt match the PathVariable
      * @throws URISyntaxException if the Location URI syntax is incorrect
      */
     @PostMapping("courses/{courseId}/student-questions")
     @PreAuthorize("hasRole('USER')")
     public ResponseEntity<StudentQuestion> createStudentQuestion(@PathVariable Long courseId, @RequestBody StudentQuestion studentQuestion) throws URISyntaxException {
+        if (!studentQuestion.getCourse().getId().equals(courseId)) {
+            return badRequest("courseId", "400", "PathVariable courseId doesnt match the courseId of the sent StudentQuestion in Body");
```

Comment:
check for other `doesnt` misspellings in your code ;)
```suggestion
            return badRequest("courseId", "400", "PathVariable courseId doesn't match the courseId of the sent StudentQuestion in Body");
```